### PR TITLE
Refactor use map_err

### PR DESCRIPTION
**Pull request by wim-web**
issue: https://github.com/rust-bitcoin/rust-bitcoin/issues/793

change to using map_err

### CODE REVIEW

**Review by dr-orlovsky — COMMENTED** [review]
I believe there was a second place in that PR when the similar matching was happening?

File: src/util/psbt/serialize.rs [hunk 1]
```diff
@@ -205,16 +205,17 @@ impl Serialize for schnorr::SchnorrSig  {
 
 impl Deserialize for schnorr::SchnorrSig {
     fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
-        match schnorr::SchnorrSig::from_slice(&bytes) {
-            Ok(sig) => Ok(sig),
-            Err(schnorr::SchnorrSigError::InvalidSighashType(flag)) => {
-                Err(encode::Error::from(psbt::Error::NonStandardSigHashType(flag as u32)))
+        schnorr::SchnorrSig::from_slice(&bytes).map_err(|e| match e {
```
Comment (Kixunil) [inline src/util/psbt/serialize.rs:208]: I feel that one of these styles would be easier to understand by readers:

```rust
schnorr::SchnorrSig::from_slice(&bytes)
    .map_err(|e| match e {
       ...
    })
```

```rust
schnorr::SchnorrSig::from_slice(&bytes)
    .map_err(|e|
        match e {
            ...
        }
    )
```

```rust
schnorr::SchnorrSig::from_slice(&bytes)
    .map_err(|e|
        match e {
           ...
        }
    )
```

**Review by Kixunil — COMMENTED** [review]
@dr-orlovsky yes, there was another place

File: src/util/psbt/serialize.rs (hunk 1 above)
Comment (wim-web) [reply src/util/psbt/serialize.rs]: thank you for review.
fixed code!

**Comment by wim-web** [comment]
i have refactor another place too

**Comment by Kixunil** [comment]
I approved run but unless I'm too confused something it'll fail on missing commas.

**Review by Kixunil — APPROVED** [review]
ACK 9f848472e4b49111257d97816d2ad81572064c7b

Thank you!

**Review by apoelstra — APPROVED** [review]
ACK 9f848472e4b49111257d97816d2ad81572064c7b

**Pull request merged**